tools/dev-doctor/main.go

```diff
@@ -0,0 +1,121 @@
+// Command doctor checks the development setup for common problems.
+package main
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"text/tabwriter"
+
+	"github.com/blang/semver"
+)
+
+// A checkResult is the result of a check.
+type checkResult int
+
+const (
+	checkSkipped checkResult = -1 // The check was skipped.
+	checkOK      checkResult = 0  // The check completed and did not find any problems.
+	checkWarning checkResult = 1  // The check completed and found something that might indicate a problem.
+	checkError   checkResult = 2  // The check completed and found a problem.
+	checkFailed  checkResult = 3  // The check could not be completed.
+)
+
+// A check is an individual check.
+type check interface {
+	id() string
+	run() (checkResult, string)
+}
+
+var checkResultStr = map[checkResult]string{
+	checkSkipped: "skipped",
+	checkOK:      "ok",
+	checkWarning: "warning",
+	checkError:   "error",
+	checkFailed:  "failed",
+}
+
+func main() {
+	checks := []check{
+		osArchCheck{},
+		&binaryCheck{
+			name:          "go",
+			versionArgs:   []string{"version"},
+			versionRegexp: regexp.MustCompile(`go version go(\d+\.\d+\.\d+)`),
+			minVersion:    &semver.Version{Major: 1, Minor: 15, Patch: 0},
+		},
+		&binaryCheck{
+			name:          "clang",
+			versionArgs:   []string{"--version"},
+			versionRegexp: regexp.MustCompile(`clang version (\d+\.\d+\.\d+)`),
+			minVersion:    &semver.Version{Major: 10, Minor: 0, Patch: 0},
+		},
+		// FIXME add llvm check?
+		// FIXME add libelf-devel check?
+		&binaryCheck{
+			name:          "ginkgo",
+			versionArgs:   []string{"version"},
+			versionRegexp: regexp.MustCompile(`Ginkgo Version (\d+\.\d+\.\d+)`),
+			minVersion:    &semver.Version{Major: 1, Minor: 4, Patch: 0},
+		},
+		// FIXME add gomega check?
+		&binaryCheck{
+			name:          "golangci-lint",
+			versionArgs:   []string{"--version"},
```

Comment:
Weird, this flag is unknown for my `golangci-lint` and `version` gives no information. I can double check how I installed it if that helps.
That being said, do we really need a version check or did we include the version more as a safeguard? /cc @sayboras 
```
$ golangci-lint --version
Error: unknown flag: --version
[...]
$ golangci-lint version
golangci-lint has version master built from ? on
```

Reply:
With the current configuration in .golangci.yaml, it's more as a safeguard :)

I just found out yesterday that `--version` flag only available in release (i.e. building from source will not have it)

Reply:
OK, I've switched to using `golangci-lint version` instead.